File: 2024/aquecimento1/python/varetas.py
valores_varetas = [5, 10, 15, 20, 50]

cont_rodadas = 1
while True:
    jogador_atual = int(input())
    if jogador_atual == -1:
        break

    jogador_atual -= 1
    pontuacoes  = [0, 0, 0]
    for vareta_pega in [int(n) for n in input().split(" ") if n != "-1"]:
        if vareta_pega == 0:
            jogador_atual = (jogador_atual + 1) % 3
            continue

        pontuacoes[jogador_atual] += valores_varetas[vareta_pega - 1]
    
    vencedores = []
    for i in range(3):
        if pontuacoes[i] == max(pontuacoes):
            vencedores.append(i + 1)

    if (cont_rodadas > 1):
        print()

    print(f"RODADA {cont_rodadas}")
    print(f"{'Ganhador' if len(vencedores) == 1 else 'Empate'} com {max(pontuacoes)} pontos")
    for i in range(len(vencedores)):
        print(f"{', ' if i > 0 else ''}Jogador {vencedores[i]}", end="")
    print()

    cont_rodadas += 1

File: 2024/aquecimento1/C/varetas.c
#include <stdio.h>

#define true 1
const int VALORES_VARETAS[5] = {5, 10, 15, 20, 50};

int main(void) {
    int cont_rodadas = 1;

    while (true) {
        int jogador_atual;
        scanf("%d", &jogador_atual);
        if (jogador_atual == -1) {
            break;
        }

        jogador_atual--;
        int pontuacoes[3] = {0, 0, 0};
        while (true) {
            int vareta_retirada;
            scanf("%d", &vareta_retirada);
            if (vareta_retirada == -1) {
                break;
            }

            if (vareta_retirada == 0) {
                jogador_atual = (jogador_atual + 1) % 3;
                continue;
            }

            pontuacoes[jogador_atual] += VALORES_VARETAS[vareta_retirada - 1];
        }

        int maior_pontuacao = pontuacoes[0];
        for (int jogador = 0; jogador < 3; jogador++) {
            maior_pontuacao = (pontuacoes[jogador] > maior_pontuacao) ? pontuacoes[jogador] : maior_pontuacao;
        }
        
        int vencedores[3] = {0, 0, 0};
        int cont_vencedores = 0;
        for (int jogador = 0; jogador < 3; jogador++) {
            if (pontuacoes[jogador] == maior_pontuacao) {
                vencedores[cont_vencedores] = jogador + 1;
                cont_vencedores++;
            }
        }
        
        if (cont_rodadas > 1) {
            printf("\n");
        }
        printf("RODADA %d\n", cont_rodadas);
        printf("%s com %d pontos\n", ((vencedores[1] == 0) ? "Ganhador" : "Empate"), maior_pontuacao);
        printf("Jogador %d", vencedores[0]);
        for (int jogador = 1; vencedores[jogador] != 0; jogador++) {
            printf(", Jogador %d", vencedores[jogador]);
        }
        printf("\n");

        cont_rodadas++;
    }

    return 0;
}